RecoHI/HiJetAlgos/plugins/HiPuRhoProducer.cc

```diff
@@ -0,0 +1,645 @@
+
+  fjInputs_ = fjOriginalInputs_;
+
+  std::vector<int> jettowers;                       // vector of towers indexed by "user_index"
+  std::vector<std::pair<int, int>> excludedTowers;  // vector of excluded ieta, iphi values
+
+  int32_t nref = 0;
+
+  for (auto const& pseudojetTMP : fjJets_) {
+    if (nref < nMaxJets_) {
+      jtexngeom[nref] = 0;
+      jtexntow[nref] = 0;
+      jtexpt[nref] = 0;
+      jtpt[nref] = pseudojetTMP.perp();
+      jteta[nref] = pseudojetTMP.eta();
+      jtphi[nref] = pseudojetTMP.phi();
+    }
+
+    if (pseudojetTMP.perp() < puPtMin_)
+      continue;
+
+    for (auto const& im : allgeomid_) {
+      double dr = reco::deltaR(geo_->getPosition((DetId)im), pseudojetTMP);
+      std::vector<std::pair<int, int>>::const_iterator exclude =
+          std::find(excludedTowers.begin(), excludedTowers.end(), std::pair<int, int>(im.ieta(), im.iphi()));
+      if (dr < radiusPU_ && exclude == excludedTowers.end() &&
+          (geomtowers_[im.ieta()] - ntowersWithJets_[im.ieta()]) > minimumTowersFraction_ * (geomtowers_[im.ieta()])) {
+        ntowersWithJets_[im.ieta()]++;
+        excludedTowers.emplace_back(std::pair<int, int>(im.ieta(), im.iphi()));
+
+        if (nref < nMaxJets_)
+          jtexngeom[nref]++;
+      }
+    }
+
+    for (auto const& input : fjInputs_) {
+      int index = input.user_index();
+      const reco::CandidatePtr& originalTower = inputs_[index];
+
+      int ie = ieta(originalTower);
+      int ip = iphi(originalTower);
+      auto exclude = std::find(excludedTowers.begin(), excludedTowers.end(), std::pair<int, int>(ie, ip));
+      if (exclude != excludedTowers.end()) {
+        jettowers.push_back(index);
+      }
+
+      double dr = reco::deltaR(input, pseudojetTMP);
+      if (dr < radiusPU_ && nref < nMaxJets_) {
+        jtexntow[nref]++;
+        jtexpt[nref] += originalTower->pt();
+      }
+    }
+
+    if (nref < nMaxJets_)
+      nref++;
+  }
+
+  // Create a new collections from the towers not included in jets
+  for (auto const& input : fjInputs_) {
+    int index = input.user_index();
+    const reco::CandidatePtr& originalTower = inputs_[index];
+    auto itjet = std::find(jettowers.begin(), jettowers.end(), index);
+    if (itjet == jettowers.end()) {
+      orphanInput.emplace_back(originalTower->px(), originalTower->py(), originalTower->pz(), originalTower->energy());
+      orphanInput.back().set_user_index(index);
+    } else {
+      towExcludePt_.push_back(originalTower->pt());
+      towExcludePhi_.push_back(originalTower->phi());
+      towExcludeEta_.push_back(originalTower->eta());
+    }
+  }
+
+  postOrphan_ = true;
+}
+
+void HiPuRhoProducer::putRho(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  std::size_t size = etaEdgeLow_.size();
+
+  std::vector<std::pair<std::size_t, double>> order;
+  for (std::size_t i = 0; i < size; ++i) {
+    order.emplace_back(std::make_pair(i, etaEdgeLow_[i]));
```

Comment:
```suggestion
    order.emplace_back(i, etaEdgeLow_[i]);
```
no need for details with emplace